### PR TITLE
fix(codex): remove gentle-dev permission profile, rely on Codex defaults

### LINKED ISSUE #1398: fix(codex): remove restrictive gentle-dev permission profile, rely on Codex defaults
### Pre-flight Checklist

- [x] I have searched [existing issues](https://github.com/Gentleman-Programming/gentle-ai/issues) and this is not a duplicate
- [x] I understand that PRs will be rejected if the linked issue does not have `status:approved`

### 📝 Bug Description

Gentle AI's Codex permission injection writes a custom restrictive permission profile into `~/.codex/config.toml` (`internal/components/permissions/inject.go` `injectCodexPermissions`): it sets `approval_policy = "on-request"`, `default_permissions = "gentle-dev"`, and a full `permissions.gentle-dev` table whose filesystem baseline is `":minimal" = "read"` plus a short explicit allowlist (git config, Nix/Home Manager paths, tmpdir writes, workspace roots).

Under the unified ChatGPT/Codex app sandbox on macOS, that `:minimal` read baseline blocks everything not explicitly listed, which breaks core development toolchains:

- Apple Developer Tools: `xcode-select`, `xcrun`, `/usr/bin/git` (shims under `/usr/bin` and `/Library/Developer/CommandLineTools` are not in the read allowlist).
- Global OpenSSL configuration reads (`/etc/ssl`, `/opt/homebrew/etc/openssl*`).
- `npm` wrapper execution (reads outside the allowlist during startup).

Reported by a community user on Gentle AI 2.1.7, macOS Golden Gate 27.0 Beta 3, Codex app after the ChatGPT unification: the repository itself is writable and Codex's bundled Git works, but the `gentle-dev` profile degrades everything else. The same environment worked before the unification because the previous app did not enforce the injected profile the same way.

Codex's own default permission behavior (no `default_permissions` override) allows broad read access with workspace-scoped writes, which is compatible with these toolchains.

### 🔄 Steps to Reproduce

1. Install gentle-ai 2.1.7 with the Codex agent on macOS and let it inject permissions (`injectCodexPermissions`).
2. Open the unified Codex app on a repository; the injected `default_permissions = "gentle-dev"` profile activates.
3. Run `xcrun --version`, `git --version` (Apple git), or `npm install` inside the session.

### ✅ Expected Behavior

Gentle AI should not pin a custom restrictive sandbox profile for Codex. Injection should rely on Codex's built-in default permissions:

- Stop writing `default_permissions = "gentle-dev"`, the `permissions.gentle-dev` table, and the injected `approval_policy` override.
- Migration: on the next inject/install run, REMOVE those previously-injected keys/tables from an existing `~/.codex/config.toml` (only the exact keys gentle-ai wrote; user-authored content must remain untouched), idempotently.
- Uninstall path tracking and tests updated accordingly.

### ❌ Actual Behavior

The injected `gentle-dev` profile with a `:minimal` filesystem read baseline blocks Apple Developer Tools, OpenSSL config reads, and npm wrappers inside the Codex app sandbox, degrading the default development experience.

### Operating System

macOS

### Agent / Client

Codex

### Shell

zsh

### Additional Context

Source: `internal/components/permissions/inject.go:179-253` (`injectCodexPermissions`), reached via `Inject` for `model.AgentCodex`. The SDD profiles in `internal/agents/codex/profiles.go` only set model/effort and are unaffected.

### PR DESCRIPTION
**Pull request by Alan-TheGentleman**
## Summary

Fixes #1398: the injected `permissions.gentle-dev` profile (with a `:minimal` filesystem read baseline) broke Apple Developer Tools (`xcrun`, `xcode-select`, `/usr/bin/git`), global OpenSSL config reads, and npm wrappers under the unified Codex app sandbox on macOS. Per maintainer decision, gentle-ai no longer pins a custom Codex sandbox profile — Codex's built-in defaults govern.

- `injectCodexPermissions` is now a migration-only cleanup: removes `approval_policy` only when it is exactly the injected `"on-request"`, `default_permissions` only when `"gentle-dev"`, and the entire `permissions.gentle-dev` table tree. User-authored content (custom values, other `permissions.*` profiles) is preserved byte-for-byte; idempotent; never creates the file.
- New quote-aware `filemerge` helpers (`RemoveTOMLTableTree`, `RemoveTopLevelTOMLKeyIfValue`) match TOML key paths segment-wise — quoted headers/keys (`[permissions.\"gentle-dev\"]`) are matched, prefix collisions (`gentle-devtools`) structurally cannot occur, and a malformed header makes table context unknown so removal never guesses (the bounded review caught and we fixed a data-loss edge there, pinned by regression tests).
- Backup/rollback coverage follows the mutation: `CleanupPath` is included in the Permission component's backup targets whenever the file's absence is not positively confirmed, while post-apply verification never requires an absent file.
- The old injector, its Windows migration, and its in-package TOML parser are deleted (~980 lines); the parser's quote-aware core was resurrected into `filemerge` intact.

Known residual (review-documented): backup targets are fixed at plan time, so a file created by a concurrent external process between plan and apply is mutated without backup — reachable only via genuine cross-process concurrency; the backup snapshotter independently fails closed on ambiguous stat errors.

Workaround for affected users until release: delete `default_permissions = \"gentle-dev\"` and the `[permissions.gentle-dev]` tree from `~/.codex/config.toml` and restart the app — or just run the new version's sync, which does exactly that.

## Review

Three-round 4R bounded review: round 1 caught a CRITICAL (quoted TOML forms silently surviving cleanup), round 2 caught a CRITICAL (stale table context deleting user content after a malformed header) — both fixed TDD-first with verbatim-repro regression tests.

Refs #1398

## Summary by CodeRabbit

- **Bug Fixes**
  - During both normal and deferred synchronization, the retired Codex `gentle-dev` permissions profile is now removed instead of repaired/added.
  - Cleanup is repeatable (idempotent) and preserves existing user-authored Codex configuration.

- **Improvements**
  - TOML modifications are safer: targeted subtree and conditional top-level key removal avoid unintended edits.
  - Component-path verification and sync/override behavior are more reliable, including correct handling of override semantics and persisted selections.

- **Tests**
  - Expanded regression coverage for Codex cleanup/deferral, verification target selection, TOML removal edge cases, and sync/override persistence.

### CODE REVIEW

File: internal/components/permissions/inject.go [hunk 1]
```diff
@@ -3,38 +3,41 @@ package permissions
...
 		return ""
 	}
 	return adapter.SettingsPath(homeDir)
 }
 
+// CleanupPath returns the file that permission cleanup may modify for the
+// adapter without ever creating it, or an empty string when the agent has no
+// cleanup target. Unlike TargetPath it is not an injection output: callers
+// that snapshot files for backup/rollback should include it only when it
+// already exists, and it must never be treated as a required post-apply file.
```
[the remark this answers is not in the document]
Comment (Alan-TheGentleman) [reply internal/components/permissions/inject.go]: Addressed in 6a470255. CleanupPath now tells backup callers to include the path when it exists or when its absence cannot be confirmed, matching the fail-closed stat behavior.

File: internal/components/permissions/inject.go [hunk 2]
```diff
@@ -179,263 +183,43 @@ func Inject(homeDir string, adapter agents.Adapter) (InjectionResult, error) {
...
-	merged = filemerge.UpsertTOMLTableKey(merged, "permissions.gentle-dev.filesystem", "glob_scan_max_depth", "6")
-	for _, path := range []string{`"."`, `".git/**"`} {
-		merged = filemerge.UpsertTOMLTableKey(merged, `permissions.gentle-dev.filesystem.":workspace_roots"`, path, `"write"`)
-	}
-	for _, path := range secretDenyPaths {
-		merged = filemerge.UpsertTOMLTableKey(merged, `permissions.gentle-dev.filesystem.":workspace_roots"`, path, `"deny"`)
+	if baseTOML == nil {
+		// No config file: nothing to clean, and cleanup never creates one.
+		return InjectionResult{}, nil
 	}
```
[the remark this answers is not in the document]
Comment (Alan-TheGentleman) [reply internal/components/permissions/inject.go:207]: Addressed in 6a470255. Codex cleanup now reads config.toml directly, keeps ENOENT as a no-op, and wraps other failures with TOML-specific context while preserving the original path error.

File: internal/components/filemerge/toml.go [hunk 3]
```diff
@@ -317,6 +322,235 @@ func RemoveTOMLTable(content, tableName string) string {
...
+		return content
+	}
+
+	lines := strings.SplitAfter(content, "\n")
+	kept := make([]string, 0, len(lines))
+	var tablePath []string
+	tableKnown := true
+	removing := false
+	for _, line := range lines {
+		if strings.HasPrefix(strings.TrimSpace(line), "[") {
```
[the remark this answers is not in the document]
Comment (Alan-TheGentleman) [reply internal/components/filemerge/toml.go]: Addressed in 6a470255. The cleanup tracks multiline basic and literal string state, rejects Go-only quoted-key escapes, and removes targeted dotted assignments through the closing multiline delimiter. Focused regressions preserve unrelated multiline content and headers.

File: internal/components/filemerge/toml.go [hunk 4]
```diff
@@ -317,6 +322,341 @@ func RemoveTOMLTable(content, tableName string) string {
...
+		}
+		if removing {
+			continue
+		}
+		if tableKnown {
+			if keyPath, isAssignment := parseTOMLAssignmentKey(line); isAssignment {
+				fullPath := append(append([]string(nil), tablePath...), keyPath...)
+				if hasTOMLKeyPathPrefix(fullPath, target) {
+					removingMultilineValue = multilineQuote != 0
+					continue
```
[the remark this answers is not in the document]
Comment (Alan-TheGentleman) [reply internal/components/filemerge/toml.go:394]: Addressed in de5190b2. TestRemoveTOMLTableTreeRemovesMultilineArrayAssignment first reproduced the orphaned array body, then proved complete removal across nested arrays while ignoring brackets inside single-line strings, multiline strings, and comments and preserving an unrelated array plus the following table. Both the original and cleaned fixtures parse as TOML, and the focused packages, sync cleanup tests, and full Go suite pass.

File: internal/components/filemerge/toml.go [hunk 5]
```diff
@@ -317,6 +322,366 @@ func RemoveTOMLTable(content, tableName string) string {
...
+	kept := make([]string, 0, len(lines))
+	topLevel := true
+	var multilineQuote byte
+	for _, line := range lines {
+		insideMultiline := multilineQuote != 0
+		advanceTOMLMultilineState(line, &multilineQuote)
+		if topLevel && !insideMultiline {
+			if strings.HasPrefix(strings.TrimSpace(line), "[") {
+				topLevel = false
+			} else if equals := tomlIndexOutsideQuotes(line, '='); equals != -1 {
```
[the remark this answers is not in the document]
Comment (Alan-TheGentleman) [reply internal/components/filemerge/toml.go]: Addressed in cd254bd5. TestRemoveTopLevelTOMLKeysAfterNestedArray first reproduced both legacy keys surviving after a valid nested top-level array, then proved exact approval_policy and default_permissions removal while preserving bracket text in strings/comments, the unrelated array, and identical keys under a real following table. The focused packages, sync cleanup tests, full Go suite, and TOML parser validation pass.

**Comment by Alan-TheGentleman** [comment]
Superseded by #1416, which merged the same approved candidate as `cddb1ae8b3293eb07b4ec092bd18318c8956f299` from a clean branch rooted at current `main`.

We used the replacement instead of force-pushing this PR because this branch's merge history made native publication-range validation ambiguous through duplicate paths. Closing #1404 without merge; #1398 is resolved by #1416.
